### ISSUE TITLE
Support existing and new Analyzer output formats

### ISSUE THREAD

**Issue by kuhnroyal** [insideapp-oss/sonar-flutter#51]
This is a follow-up to #47

There is now a `--format=machine` flag for `dartanalyzer` and `dart analyze`, both produce the same output
```
INFO|HINT|UNUSED_ELEMENT|/Users/.../example/lib/src/example_form.dart|8|7|4|The declaration '_foo' isn't referenced.
INFO|LINT|PREFER_SINGLE_QUOTES|/Users/.../example/lib/src/example_form.dart|8|14|2|Only use double quotes for strings containing single quotes.
```
* this is good, we should probably support this as the new default and also update the readme 
* `flutter analyze` does not support this flag yet, but there is https://github.com/flutter/flutter/projects/106 with several issue about adding the support

Without the `--format=machine` flag there are 3 differently formatted outputs for `flutter analyze`, `dart analyze` and `dartanalyzer`, probably also depending on the version of Dart/Flutter.

There is currently no error when the format can not be parsed, it just results in zero issues in Sonar, this causes frustration and should be fixed. 

FYI @mreichelt if you have any ideas

**Comment by kuhnroyal** [comment]
**NOT SUPPORTED** Output for `flutter analyze --write=analysis-results.txt` with Flutter 2.0.5:
```
[info] The declaration '_foo' isn't referenced (/Users/.../example/lib/src/example_form.dart:8:7)
[info] Only use double quotes for strings containing single quotes (/Users/.../example/lib/src/example_form.dart:8:14)
```

**NOT SUPPORTED** Output for `dart analyze --format=machine > analysis-results.txt` (same for `dartanalyzer --format=machine`):
```
INFO|HINT|UNUSED_ELEMENT|/Users/.../example/lib/src/example_form.dart|8|7|4|The declaration '_foo' isn't referenced.
INFO|LINT|PREFER_SINGLE_QUOTES|/Users/.../example/lib/src/example_form.dart|8|14|2|Only use double quotes for strings containing single quotes.
```

**SUPPORTED?** Output for `flutter analyze > analysis-results.txt` with Flutter 2.0.5:
```
Analyzing example...                                            

   info • The declaration '_foo' isn't referenced • lib/src/example_form.dart:8:7 • unused_element
   info • Only use double quotes for strings containing single quotes • lib/src/example_form.dart:8:14 • prefer_single_quotes

```

**Comment by kuhnroyal** [comment]
There now is also a JSON output in the Dart SDK: https://github.com/dart-lang/sdk/blob/2371d746646843fe9688f2a716a7b125364a2fbb/pkg/dartdev/lib/src/commands/analyze.dart#L57

**Comment by kuhnroyal** [comment]
Ping, still have this on my roadmap.
